### PR TITLE
Fixing bug where the worker thread would hang if there wasn't a tslint.json file

### PR DESCRIPTION
**Pull request by Xapphire13**
Two issues in one:

**Firstly**, there was a regression I caused when copying the `rulesDirectory` code over to worker.js in [#158](https://github.com/AtomLinter/linter-tslint/pull/158). I missed the fact it used to refer to `this.` (which is now `config.`), so a circular reference was caused, which led `tslint` to have kittens, leading to the worker hanging.

**Secondly**, if `tslint` hit an issue, it would cause the worker to hang.

### CODE REVIEW

File: lib/worker.js [hunk 1]
```diff
@@ -133,8 +133,13 @@ async function lint(content, filePath, options) {
...
   }, options));
 
-  linter.lint(filePath, content, configuration);
-  const lintResult = linter.getResult();
+  let lintResult;
+  try {
+    linter.lint(filePath, content, configuration);
+    lintResult = linter.getResult();
+  } catch (err) {
+    lintResult = {};
```
Comment (Arcanemagus) [inline lib/worker.js]: Shouldn't something be done with this crash? (Logging to the console?)

File: spec/linter-tslint-spec.js [hunk 2]
```diff
@@ -47,4 +48,10 @@ describe('The TSLint provider for Linter', () => {
       }),
     );
   });
+
+  it('validates even when there is no tslint.json', () => {
+    waitsForPromise(() =>
+      atom.workspace.open(noConfigPath).then(editor => lint(editor)),
```
Comment (Arcanemagus) [inline spec/linter-tslint-spec.js]: Should the results be checked?

File: lib/worker.js (hunk 1 above)
Comment (Xapphire13) [inline lib/worker.js]: Yep, will add =]

File: spec/linter-tslint-spec.js (hunk 2 above)
Comment (Xapphire13) [inline spec/linter-tslint-spec.js]: Not really sure. In this test I don't care that the file was accurately validated, nor the specific results, I just want to make sure that validation completed

Comment (Arcanemagus) [inline spec/linter-tslint-spec.js]: If that's all that was needed for the test, that's good enough for me.

**Pull request merged**